datasets/doc/source/how-to-use-partitioners.ipynb

```diff
@@ -0,0 +1,257 @@
+   "source": [
+    "# How to use `Partitioner`s\n",
+    "\n",
+    "The aim of this tutorial is to make you familiar with the available `Partitioner`s that `Flower Datasets` have out-of-the-box."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# What is `Partitioner`?\n",
+    "\n",
+    "`Partitioner` is an object responsible for dividing a dataset according to a chosen strategy. There are many `Partitioner`s that you can use and all of them inherit from the `Partitioner` object which is an abstract class providing basic structure and methods that need to be implemented for any new `Partitioner` to integrate with the rest of `Flower Datasets` code. The creation of different `Partitioner` differs but the behavior is the same = they produce the same type of objects.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## IidPartitioner Creation\n",
+    "\n",
+    "Let's create (instantiate) the most basic partitioner, `IidPartitioner` and learn how it interacts with `FederatedDataset`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from flwr_datasets.partitioner import IidPartitioner\n",
+    "\n",
+    "partitioner = IidPartitioner(num_partitions=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Right now the partitioner does not have access to any data therefore it has nothing to partition. `FederatedDataset` is responsible for assigning data to a `partitioner`(s).\n",
+    "\n",
+    "What **part** of the data is assigned to partitioner?\n",
+    "\n",
+    "In centralized (traditional) ML, there exist a strong concept of the splits of the dataset. Typically you can hear about train/valid/test splits. In FL research, if we don't have an already divided datasets (e.g. by `user_id`), we simulate such division of a centralized dataset. The goal of that operation is to resemble an FL scenario where the data is spread around the clients. In Flower Datasets you decide what split of the dataset will be partitioned. You can also resplit the datasets such that you use a more non-custom split, or merge the whole train and test split into a single dataset but that's not a part of this tutorial.\n",
+    "\n",
+    "Let's see how you specify the split for partitioning."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How do you specify the split to partition?\n",
+    "\n",
+    "The specification of the split happens as you specify the `partitioners` argument for `FederatedDataset`. It maps `partition_id: str` to the partitioner that will be used for that split of the data. In the example below we're using `train` split of the `cifar10` dataset to partition.\n",
+    "\n",
+    "(If you're unsure why/how we chose the name of the `dataset` and how to customize it, see the first tutorial.)"
```

Comment:
Let's add a link to that "first tutorial"?

Reply:
Done